examples/dynamoathenamigration/migration.go

```diff
@@ -486,18 +489,251 @@ func (t *task) downloadFromS3(ctx context.Context, key string) (*os.File, error)
 	}
 	path := path.Join(dir, originalName)
 
-	f, err := os.Create(path)
+	originalFile, err := os.Create(path)
 	if err != nil {
 		return nil, trace.Wrap(err)
 	}
-	if _, err := t.s3Downloader.Download(ctx, f, &s3.GetObjectInput{
+	if _, err := t.s3Downloader.Download(ctx, originalFile, &s3.GetObjectInput{
 		Bucket: aws.String(t.Bucket),
 		Key:    aws.String(key),
 	}); err != nil {
-		f.Close()
+		originalFile.Close()
+		return nil, trace.Wrap(err)
+	}
+
+	defer originalFile.Close()
+
+	// maxMemoryUsedForSortingExportInBytes defines how big chunks of audit events
+	// will be loaded into memory, sorted and stored in temporary files.
+	maxMemoryUsedForSortingExportInBytes := 1024 * 1024 * t.MaxMemoryUsedForSortingExportInMB
+	sortedFile, err := createSortedExport(originalFile, dir, maxMemoryUsedForSortingExportInBytes)
+	if err != nil {
+		return nil, trace.NewAggregate(err, sortedFile.Close())
```

Comment:
Shouldn't we check if `sortedFile` is not nil? Calling close on nil file will give an error and looks like there are plenty of cases when we return an error from `createSortedExport` before even trying to open the file.

Reply:
good catch, in https://github.com/gravitational/teleport/pull/29627/commits/0d65492128cd28b279b4c4c7cb1927fe374c508a i have removed it, because it's check and closed in layer above if there is no error, and file is returned as last operation